packages/common/driver-definitions/src/storage.ts

```diff
@@ -158,8 +158,11 @@ export interface IDocumentStorageService extends Partial<IDisposable> {
 	 * in debugging purposes to see why this call was made.
 	 */
 	// TODO: use `undefined` instead.
-	// eslint-disable-next-line @rushstack/no-new-null
-	getSnapshotTree(version?: IVersion, scenarioName?: string): Promise<ISnapshotTree | null>;
+	getSnapshotTree(
+		version?: IVersion,
+		scenarioName?: string,
+		// eslint-disable-next-line @rushstack/no-new-null
+	): Promise<ISnapshotTree | IPartialSnapshotWithContents | null>;
```

Comment:
I do not remember all the history here (would need to look at code), but I wonder - can we get rid of null here? I.e. are there are any cases where snapshot is missing? And if it's only due to error cases, we should throw an exception, not return null.
Maybe it's too much for one change, but every time we change signature, we should look for opportunities to make it simpler :)

Reply:
Made IPartialSnapshotWithContents extend from ISnapshotTree. So, no need to change the type here. We can have some other task to see if we can remove null from this api.